### PR TITLE
feat(ng-lib): use GuardsCheckEnd instead of Navigationend. see #1529

### PR DESCRIPTION
**Pull request by SanderElias**
ISSUES CLOSED: #1529

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/scullyio/scully/blob/main/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: .github/test-build/action.yml [hunk 1]
```diff
@@ -71,11 +71,11 @@ runs:
         echo "install Scully"
         npx ng g @scullyio/init:ng-add --renderer puppeteer
         npx ng g @scullyio/init:blog
-        echo "build"
     - name: build and run
       shell: bash
       run: |
         cd ../demo
+        npx browserslist "> 10%" > .browserslistrc
```
Comment (Jefiozie) [inline .github/test-build/action.yml:78]: let's add a comment with the GH issue added.

File: libs/ng-lib/src/lib/transfer-state/transfer-state.service.ts [hunk 2]
```diff
@@ -57,11 +57,11 @@ export class TransferStateService {
     /** prevent emitting before navigation to _this_ URL is done. */
     switchMap((e: NavigationStart) =>
       this.router.events.pipe(
-        filter((ev) => ev instanceof NavigationEnd && ev.url === e.url),
+        filter((ev) => ev instanceof GuardsCheckEnd && ev.url === e.url),
```
Comment (Jefiozie) [inline libs/ng-lib/src/lib/transfer-state/transfer-state.service.ts:60]: Should we add a test around this for future purposes that when we change it to another `instance` we have a failing test?

**Review by Jefiozie — CHANGES_REQUESTED** [review]
Some small comments

File: libs/ng-lib/src/lib/transfer-state/transfer-state.service.ts (hunk 2 above)
Comment (SanderElias) [reply libs/ng-lib/src/lib/transfer-state/transfer-state.service.ts:60]: That is actually a good idea. I'll add a route that uses transferstae in a resolver to our kitchen sick.

**Review by Jefiozie — APPROVED** [review]
LGTM

**Pull request merged**